cell_classification/augmentation_pipeline.py

```diff
@@ -137,3 +138,351 @@ def py_aug(batch, tf_aug):
+                The minimum zoom factor
+            max_zoom (float):
+                The maximum zoom factor
+        """
+        super(Zoom, self).__init__()
+        self.prob = prob
+        self.min_zoom = min_zoom
+        self.max_zoom = max_zoom
+        self.fill_mode = fill_mode
+
+    def __call__(self, image, labels):
+        """ Applies zoom
+        Args:
+            image (tf.Tensor):
+                The image to zoom
+            labels (tf.Tensor):
+                The labels
+        Returns:
+            tf.Tensor:
+                The zoomed image
+            tf.Tensor:
+                The labels
+        """
+        if tf.random.uniform(()) < self.prob:
+            zoom_factor = tf.random.uniform([1], self.min_zoom, self.max_zoom)
+            zoom_factor = 1/zoom_factor
+            zooms = tf.expand_dims(
+                tf.cast(
+                    tf.concat([zoom_factor, zoom_factor], axis=0), dtype=tf.float32
+                ), 0
+            )
+            squeeze = False
+            if tf.rank(image) == 3:
+                image = tf.expand_dims(image, axis=0)
+                labels = tf.expand_dims(labels, axis=0)
+                squeeze = True
+            _, h, w, _ = image.shape
+            image = transform(
+                image,
+                get_zoom_matrix(zooms, h, w),
+                fill_mode=self.fill_mode,
+                interpolation="bilinear",
+            )
+            labels = transform(
+                labels,
+                get_zoom_matrix(zooms, h, w),
+                fill_mode=self.fill_mode,
+                interpolation="nearest",
+            )
+            if squeeze:
+                image = tf.squeeze(image, axis=0)
+                labels = tf.squeeze(labels, axis=0)
+        return image, labels
+
+
+class LinearContrast(tf.Module):
+    """ Linear contrast augmentation"""
+    def __init__(self, prob=0.5, min_factor=0.5, max_factor=1.5):
+        """
+        Args:
+            prob (float):
+            The probability of applying the augmentation
+        min_factor (float):
+            The minimum contrast factor
+        max_factor (float):
+            The maximum contrast factor
+        """
+        super(LinearContrast, self).__init__()
+        self.prob = prob
+        self.min_factor = min_factor
+        self.max_factor = max_factor
+
+    def __call__(self, image, labels):
+        """ Apply linear contrast augmentation
+        Args:
+            image (tf.Tensor):
+                The image to augment
+                labels (tf.Tensor):
+                The labels to augment
+        Returns:
+            tf.Tensor:
+                The augmented image
+            tf.Tensor:
+                The augmented labels
+        """
+        if tf.random.uniform(()) < self.prob:
+            factor = tf.random.uniform((), self.min_factor, self.max_factor)
+            image *= factor
+        return image, labels
+
+
+class MixUp(tf.Module):
```

Comment:
Using MixUp, even without a schedule for `p` for promix, will help to avoid the issue we saw of the predictions being heavily skewed towards 0 and 1. Even with only high-confidence positives and negatives in the original training data, we'll now have targets that are in the middle regime